src/librustc_mir/transform/const_prop.rs

```diff
@@ -405,13 +405,15 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
                 }
 
                 let arg = self.eval_operand(arg, source_info)?;
+                let oflo_check = self.tcx.sess.overflow_checks();
                 let val = self.use_ecx(source_info, |this| {
                     let prim = this.ecx.read_immediate(arg)?;
                     match op {
                         UnOp::Neg => {
```

Comment:
Oh right. Now we have the original issue again. I believe this can only happen for signed integers, so add one more check and we should be good to go: `prim.layout.ty.is_signed()` (https://doc.rust-lang.org/nightly/nightly-rustc/rustc/ty/struct.TyS.html#method.is_signed) Should return `true` if the type that we're negating is a signed integer

Reply:
Is the condition also needed at the later change?

Reply:
No, the other place does a type agnostic check already. Only the bit trickery here is problematic